File: src/scheme/deposit.js
import { Deposit } from "core/domain/deposit";
import mongoose, { Schema, Document } from "mongoose";

// Define the Mongoose interface extending the Mongoose Document
export interface Deposit extends Document, Deposit {}

// Define the Mongoose schema based on the Zod schema
const DepositSchema: Schema = new Schema({
  blockNumber: { type: String, required: true },
  blockTimestamp: { type: String, required: true },
  fee: { type: String, required: false },
  hash: { type: String, required: false },
  pubkey: { type: String, required: true },
  blockchain: { type: String, required: true },
  network: { type: String, required: true },
  token: { type: String, required: true },
});

// Create and export the Mongoose model
export const DepositModel = mongoose.model<Deposit>("Deposit", DepositSchema);

File: src/main/java/com/example/Deposit.java
package com.example;

public class Deposit {
    private String blockNumber;
    private String blockTimestamp;
    private String fee;
    private String hash;
    private String pubkey;

    // Constructor
    public Deposit(String blockNumber, String blockTimestamp, String fee, String hash, String pubkey) {
        this.blockNumber = blockNumber;
        this.blockTimestamp = blockTimestamp;
        this.fee = fee;
        this.hash = hash;
        this.pubkey = pubkey;
    }

    // Getters
    public String getBlockNumber() {
        return blockNumber;
    }

    public String getBlockTimestamp() {
        return blockTimestamp;
    }

    public String getFee() {
        return fee;
    }

    public String getHash() {
        return hash;
    }

    public String getPubkey() {
        return pubkey;
    }

    @Override
    public String toString() {
        return "Deposit{" +
                "blockNumber='" + blockNumber + '\'' +
                ", blockTimestamp='" + blockTimestamp + '\'' +
                ", fee='" + fee + '\'' +
                ", hash='" + hash + '\'' +
                ", pubkey='" + pubkey + '\'' +
                '}';
    }
}
